### PR TITLE
Update LabelSet visual properties on model and model.source change 

### LINKED ISSUE #4996: LabelSet's text color not updating properly on changing column data source
LabelSet's text_color value is not being updated properly when updating the column data source using a callback. Using the code snippet below, color of the glyphs and arrow annotation seem to be changing properly but the labelset's color is not reflecting properly after filtering.

```python
from bokeh.plotting import show, output_file
from bokeh.models import Plot, Label, Circle, ColumnDataSource, DataRange1d, LinearAxis, CustomJS, LabelSet, Select, Triangle, Arrow
from bokeh.layouts import row, widgetbox

output_file('gitex.html')

xdr = DataRange1d()
ydr = DataRange1d()

p = Plot(x_range=xdr, y_range=ydr, plot_width=300, plot_height=300, background_fill_color='#F0F0F0')

source = ColumnDataSource(dict(x=[1.0, 2.0, 3.0], y=[2.0, 3.0, 1.0], text=['A', 'B', 'C'], textcol=['#FF0000','#00FF00','#0000FF']))
orig = ColumnDataSource(dict(x=[1.0, 2.0, 3.0], y=[2.0, 3.0, 1.0], text=['A', 'B', 'C'], textcol=['#FF0000','#00FF00','#0000FF']))

xaxis = LinearAxis()
p.add_layout(xaxis, 'below')

yaxis = LinearAxis()
p.add_layout(yaxis, 'left')

gly = Circle(x='x', y='y', fill_color='textcol', size=15)
p.add_glyph(source, gly)

p.add_layout(LabelSet(x='x', y='y', text='text', text_color='textcol', source=source))

gly2 = Triangle(x='x', y='y', fill_color='textcol', size=15)
p.add_glyph(source, gly2)

p.add_layout(Arrow(x_start='x', y_start='y', x_end='y', y_end='x', line_width=5, line_color='textcol', source=source))

sel = Select(title="Options:", value="All", options=["A", "B", "C", "All"])

arg_dct = dict(
    filtered_source=source,
    original_source=orig,
    element_type_select=sel
)

widget_callback_code = """
var filtered_data = filtered_source.get('data');
var original_data = original_source.get('data');

var element_type = element_type_select.get('value');

for (var key in original_data) {
    filtered_data[key] = [];
    for (var i = 0; i < original_data[key].length; ++i) {
        if ((element_type === "All" || original_data["text"][i] === element_type)) {
            filtered_data[key].push(original_data[key][i]);
        }
    console.log(key)
    console.log(filtered_data[key])
    }
}
filtered_source.trigger('change');
"""

generic_callback = CustomJS(args=arg_dct, code=widget_callback_code)
sel.callback = generic_callback

show(row(widgetbox(sel), p))
```

Discussed issue with @bryevdv here: https://groups.google.com/a/continuum.io/forum/#!topic/bokeh/P_b8ddxL2DY

### PR DESCRIPTION
**Pull request by canavandl**
- [x] issues: fixes #4996 
- [ ] tests added / passed
- [ ] release document entry (if new feature or API change)

Update LabelSet visual properties on model and model.source change

### CODE REVIEW

**Comment by canavandl** [comment]
example:

```python
from bokeh.plotting import show, output_file
from bokeh.models import Plot, Label, Circle, ColumnDataSource, DataRange1d, LinearAxis, CustomJS, LabelSet, Select, Triangle, Arrow
from bokeh.layouts import row, widgetbox

output_file('gitex.html')

xdr = DataRange1d()
ydr = DataRange1d()

p = Plot(x_range=xdr, y_range=ydr, plot_width=300, plot_height=300, background_fill_color='#F0F0F0')

source = ColumnDataSource(dict(x=[1.0, 2.0, 3.0], y=[2.0, 3.0, 1.0], text=['A', 'B', 'C'], textcol=['#FF0000','#00FF00','#0000FF']))
orig = ColumnDataSource(dict(x=[1.0, 2.0, 3.0], y=[2.0, 3.0, 1.0], text=['A', 'B', 'C'], textcol=['#FF0000','#00FF00','#0000FF']))

xaxis = LinearAxis()
p.add_layout(xaxis, 'below')

yaxis = LinearAxis()
p.add_layout(yaxis, 'left')

gly = Circle(x='x', y='y', fill_color='textcol', size=15)
p.add_glyph(source, gly)

p.add_layout(LabelSet(x='x', y='y', text='text', text_color='textcol', source=source))

gly2 = Triangle(x='x', y='y', fill_color='textcol', size=15)
p.add_glyph(source, gly2)

p.add_layout(Arrow(x_start='x', y_start='y', x_end='y', y_end='x', line_width=5, line_color='textcol', source=source))

sel = Select(title="Options:", value="All", options=["A", "B", "C", "All"])

arg_dct = dict(
    filtered_source=source,
    original_source=orig,
    element_type_select=sel
)

widget_callback_code = """
var filtered_data = filtered_source.get('data');
var original_data = original_source.get('data');

var element_type = element_type_select.get('value');

for (var key in original_data) {
    filtered_data[key] = [];
    for (var i = 0; i < original_data[key].length; ++i) {
        if ((element_type === "All" || original_data["text"][i] === element_type)) {
            filtered_data[key].push(original_data[key][i]);
        }
    console.log(key)
    console.log(filtered_data[key])
    }
}
filtered_source.trigger('change');
"""

generic_callback = CustomJS(args=arg_dct, code=widget_callback_code)
sel.callback = generic_callback

show(row(widgetbox(sel), p))
```

**Comment by canavandl** [comment]
ping @hkhare42 for review

File: bokehjs/src/coffee/models/annotations/label_set.coffee [hunk 1]
```diff
@@ -24,19 +23,23 @@ class LabelSetView extends TextAnnotation.View
...
+        @set_data()
         @plot_view.request_render())
       @listenTo(@mget('source'), 'change', () ->
-        @set_data(@mget('source'))
+        @set_data()
         @plot_view.request_render())
 
+  set_data: () ->
+    super(@mget('source'))
+    @set_visuals(@mget('source'))
```
Comment (bryevdv) [inline bokehjs/src/coffee/models/annotations/label_set.coffee]: `@source` ? It's not computed AFAIK so that should work

Comment (canavandl) [inline bokehjs/src/coffee/models/annotations/label_set.coffee]: fixed

File: bokehjs/src/coffee/models/annotations/label_set.coffee [hunk 2]
```diff
@@ -9,42 +9,45 @@ class LabelSetView extends TextAnnotation.View
...
       # dispatch CSS update immediately
       @listenTo(@model, 'change', () ->
-        @set_data(@mget('source'))
+        @set_data()
         @render())
-      @listenTo(@mget('source'), 'change', () ->
-        @set_data(@mget('source'))
+      @listenTo(@model.source, 'change', () ->
+        @set_data()
         @render())
```
Comment (bryevdv) [inline bokehjs/src/coffee/models/annotations/label_set.coffee]: I think `@plot_view.request_render()` should work, that would be better than a synchronous call to `render` if so.

File: bokehjs/src/coffee/models/annotations/label_set.coffee [hunk 3]
```diff
@@ -9,42 +9,45 @@ class LabelSetView extends TextAnnotation.View
...
         @title_div.appendTo(@$el)
 
   bind_bokeh_events: () ->
-    if @mget('render_mode') == 'css'
+    if @model.render_mode == 'css'
       # dispatch CSS update immediately
       @listenTo(@model, 'change', () ->
-        @set_data(@mget('source'))
+        @set_data()
         @render())
```
Comment (bryevdv) [inline bokehjs/src/coffee/models/annotations/label_set.coffee]: `@plot_view.request_render()` I think

**Comment by bryevdv** [comment]
Tested example working, code LGTM

**Pull request merged**